dot/network/service.go

```diff
@@ -542,13 +575,15 @@ func (s *Service) Peers() []common.PeerInfo {
 	peers := []common.PeerInfo{}
 
 	for _, p := range s.host.peers() {
-		// TODO: update this based on BlockAnnounce handshake info
+		// Load up a node/entity from peer id
+		// Query data directly
+		peerHandshake := s.notificationsProtocols[BlockAnnounceMsgType].handshakeData[p]
```

Comment:
this looks good, just make sure to do a check if it's `nil`

Reply:
Ok, the only thing is that the message I'm using is the outboundMsg which should be the same as the one I was using before